sklearn/ensemble/tests/test_voting.py

```diff
@@ -602,3 +602,29 @@ def test_voting_verbose(estimator, capsys):
 
     estimator.fit(X, y)
     assert re.match(pattern, capsys.readouterr()[0])
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    [
+        VotingRegressor(
+            estimators=[
+                ("lr", LinearRegression()),
+                ("rf", RandomForestRegressor(random_state=123)),
+            ],
+            verbose=True,
+        ),
+        VotingClassifier(
+            estimators=[
+                ("lr", LogisticRegression(random_state=123)),
+                ("rf", RandomForestClassifier(random_state=123)),
+            ],
+            verbose=True,
+        ),
+    ],
+)
+def test_voting_get_feature_names_out(estimator):
+    eclf = estimator.fit(X, y)
+    names = eclf.get_feature_names_out()
+    cls_name = estimator.__class__.__name__.lower()
+    assert_array_equal([f"{cls_name}{i}" for i in range(eclf._n_features_out)], names)
```

Comment:
I think we can have better names here, since transform returns the class labels or probabilities for each estimator. The feature names have a connection to the estimator the created them and we have names for the estimators in `VotingClassifer`.

For example for `VotingRegressor`, maybe `['lr_predict', 'rf_predict']` can be the feature names.

Reply:
Hi @thomasjpfan, do I need to revert all the changes in `voting` and `stacking` in this PR before opening separate ones for those?

Reply:
Yes.